diff --git a/src/plugins/mpi/pmix/mpi_pmix.c b/src/plugins/mpi/pmix/mpi_pmix.c
index bbb947616c..65ec00ad73 100644
--- a/src/plugins/mpi/pmix/mpi_pmix.c
+++ b/src/plugins/mpi/pmix/mpi_pmix.c
@@ -105,10 +105,17 @@ static void *_libpmix_open(void)
 #elif defined PMIXP_V2_LIBPATH
 	xstrfmtcat(full_path, "%s/", PMIXP_V2_LIBPATH);
 #endif
-	xstrfmtcat(full_path, "libpmix.so");
+	xstrfmtcat(full_path, "libpmix.so.2");
 	lib_plug = dlopen(full_path, RTLD_LAZY | RTLD_GLOBAL);
 	xfree(full_path);
 
+	if (!lib_plug) {
+		dlerror();
+		xstrfmtcat(full_path, "libpmix.so.1");
+		lib_plug = dlopen(full_path, RTLD_LAZY | RTLD_GLOBAL);
+		xfree(full_path);
+	}
+
 	if (lib_plug && (HAVE_PMIX_VER != pmixp_lib_get_version())) {
 		PMIXP_ERROR("pmi/pmix: incorrect PMIx library version loaded %d was loaded, required %d version",
 			    pmixp_lib_get_version(), (int)HAVE_PMIX_VER);
